velox/functions/sparksql/Split.cpp

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <utility>
+
+#include "velox/expression/DecodedArgs.h"
+#include "velox/expression/VectorFunction.h"
+#include "velox/expression/VectorWriters.h"
+#include "velox/functions/lib/Re2Functions.h"
+#include "velox/functions/prestosql/Utf8Utils.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+
+// split(string, delimiter[, limit]) -> array(varchar)
+//
+// Splits string on delimiter and returns an array of size at most limit.
+// delimiter is a string representing regular expression.
+// limit is an integer which controls the number of times the regex is applied.
+// By default, limit is -1.
+class Split final : public exec::VectorFunction {
+ public:
+  Split() {}
+
+  void apply(
```

Comment:
Is there any particular reason this function is not implemented using Simple Function API?

Reply:
not, so you want to convert it to simple function ?

Reply:
If there is no reason for this function to be written as Vector function, then it is better to write it as a Simple function. Simple functions are easier to implement and maintain. Thanks.

Reply:
@mbasmanova thanks,  I will do the convert, but please document this part if it is mandatory to write it as a simple function instead of using a vector function unless there is a performance improvement or another significant reason, otherwise it is easy to increase the burden on developers since we may wrongly implement vector way.
 Or if there already has a doc addressing this part please share with me. Thanks.

Reply:
> If there is no reason for this function to be written as Vector function, then it is better to write it as a Simple function. Simple functions are easier to implement and maintain.

Adding some documentation for this point: https://github.com/facebookincubator/velox/pull/10528. @mbasmanova Do you think it makes sense? Thanks.

Reply:
@rui-mo Thank you, Rui. Accepted the PR with a small comment.

Reply:
@mbasmanova done to convert to simple. @rui-mo / @mbasmanova / @PHILO-HE to help review.